### PR TITLE
Updates the What is Kubernetes page

### PR DESCRIPTION
**Pull request by shavidissa**
Fixes https://github.com/kubernetes/website/issues/14306 and fixes https://github.com/kubernetes/website/issues/12333. The content is work in progress and will be updated once all the comments are received.

### CODE REVIEW

**Comment by shavidissa** [comment]
Preview: https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/

**Comment by sftim** [comment]
@shavidissa looking at this I can't easily tell what has changed

To make that more obvious you could make two commits:
- one commit that changes the wrapping and doesn't change the the rendered page
- another commit that leaves the line wrapping as-is and changes the content

Does that make sense?

**Comment by tengqm** [comment]
I'm against the change to line wrappings. The existing wrapping works well and it is the recommended way for writing paragraphs. Merging multiple lines into a single line will make future revisions difficult to review. Ideally, we should wrap long paragraphs at a fixed column width (say 120 characters).

File: content/en/docs/concepts/overview/what-is-kubernetes.md [hunk 1]
```diff
@@ -15,196 +15,71 @@ This page is an overview of Kubernetes.
...
-
-This
-[design](https://git.k8s.io/community/contributors/design-proposals/architecture/architecture.md)
-has enabled a number of other systems to build atop Kubernetes.
+Containers are a good way to bundle and run your applications. In a production environment, you need to manage the containers that run the applications and ensure that there is no downtime. For example, if a container goes down another container needs to restart. Wouldn't it be easier if this behavior was handled by a system?
+
+That's how Kubernetes comes to the rescue! Kubernetes provides you with a framework to run distributed system resiliently. It takes care of your scaling requirements, failover, deployment patterns, and more. For example, Kubernetes can easily manage a canary deployment for your system.
+
+Kubernetes provides you with:
+* Service discovery and load balancing
```
Comment (juandiegopalomino) [inline content/en/docs/concepts/overview/what-is-kubernetes.md]: list not rendering well: https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#why-do-i-need-kubernetes-and-what-can-it-do

**Comment by shavidissa** [comment]
> It's looking good so far, but the explanation of "what it is" is a bit thin-- we have a lot context of where it came from and what it's not, but not much about the identity itself.

Good point! I will add more content in.

**Comment by shavidissa** [comment]
@juandiegopalomino: I added more content in. Can you take a look and see?

**Comment by Rajakavitha1** [comment]
@shavidissa This looks good!!!!
my two cents: For https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#what-kubernetes-is-not

We could emphasis about what Kubernetes supports rather than what it does not.......
Becasue, though we mention that Kubernetes does not.........we are also mentioning about the alternatives or the options that Kubernetes provides .....I would do thick on points that Kubernetes supports. 
I am not sure if we could change the title to " How Kubernetes supports common features in PaaS offerings" or something similar!!!!!

File: content/en/docs/concepts/overview/what-is-kubernetes.md [hunk 2]
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
...
+
+**Virtualized deployment era:**  As a solution, virtualization was introduced. It allows you to run multiple Virtual Machines (VMs) on a single physical server's CPU. Virtualization allows applications to be isolated between VMs and provides a level of security as the information of one application cannot be freely accessed by another application.
+
+Virtualization allows better utilization of resources in a physical server and allows better scalability because an application can be added or updated easily, reduces hardware costs, and much more.
+
+Each VM is a full machine running all the components, including its own operating system, on top of the virtualized hardware.
+
+**Container deployment era:** Containers are similar to VMs, but they have relaxed isolation properties to share the Operating System (OS) among the applications. Therefore, containers are considered lightweight. Similar to a VM, it has its own filesystem, CPU, memory, process space, and more. As they are decoupled from the underlying infrastructure, they are portable across clouds and OS distributions.
```
Comment (steveperry-53) [inline content/en/docs/concepts/overview/what-is-kubernetes.md]: "Similar to a VM, it ..."
Needs a change to something like:
"Similar to VMs, they ..."
"Similar to a VM, a container ..."

File: content/en/docs/concepts/overview/what-is-kubernetes.md [hunk 3]
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
...
-APIs](/docs/concepts/api-extension/custom-resources/)
-that can be targeted by a general-purpose [command-line
-tool](/docs/user-guide/kubectl-overview/).
-
-This
-[design](https://git.k8s.io/community/contributors/design-proposals/architecture/architecture.md)
-has enabled a number of other systems to build atop Kubernetes.
+Containers are a good way to bundle and run your applications. In a production environment, you need to manage the containers that run the applications and ensure that there is no downtime. For example, if a container goes down, another container needs to restart. Wouldn't it be easier if this behavior was handled by a system?
+
+That's how Kubernetes comes to the rescue! Kubernetes provides you with a framework to run distributed system resiliently. It takes care of your scaling requirements, failover, deployment patterns, and more. For example, Kubernetes can easily manage a canary deployment for your system.
```
Comment (steveperry-53) [inline content/en/docs/concepts/overview/what-is-kubernetes.md]: run distributed systems

File: content/en/docs/concepts/overview/what-is-kubernetes.md [hunk 4]
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
...
+
+Kubernetes provides you with:
+
+* **Service discovery and load balancing**  
+Kubernetes can expose containers using the DNS name or using their own IP address. If traffic to a container is high, Kubernetes is able to load balance and distribute the network traffic so that the deployment is stable.
```
Comment (steveperry-53) [inline content/en/docs/concepts/overview/what-is-kubernetes.md]: expose a container using its DNS name or using its IP address

File: content/en/docs/concepts/overview/what-is-kubernetes.md [hunk 5]
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
...
+* **Automated rollouts and rollbacks**  
+You can describe the desired state for your deployed containers using Kubernetes, and it can change the actual state to the desired state at a controlled rate. For example, you can automate Kubernetes to create new containers for your deployment, remove existing containers and adopt all their resources to the new container.
+* **Automatic bin packing**  
+Kubernetes allows you to specify how much CPU and memory (RAM) each container needs. When containers have resource requests specified, Kubernetes can make better decisions to manage the resources for containers.
+* **Self-healing**  
+Kubernetes restarts containers that fail, replaces containers, kills containers that don’t respond to your user-defined health check, and doesn’t advertise them to clients until they are ready to serve.
+* **Secret and configuration management**  
+Kubernetes lets you store and manage sensitive information, such as passwords, OAuth tokens, and ssh keys. You can deploy and update secrets and application configuration without rebuilding your image, and without exposing secrets in your stack configuration.
```
Comment (steveperry-53) [inline content/en/docs/concepts/overview/what-is-kubernetes.md]: Suggestion:
without rebuilding your container images

**Comment by steveperry-53** [comment]
@shavidissa This looks good. I left just a few minor comments.

**Comment by Rajakavitha1** [comment]
Hi,

There a good point that @sftim makes in : https://kubernetes.slack.com/archives/CGW9JHPJN/p1559665692002800

Hope this helps!!!!!

Rajie

**Comment by shavidissa** [comment]
> @shavidissa This looks good. I left just a few minor comments.

Thanks a lot for the review, @steveperry-53!

**Comment by shavidissa** [comment]
> Hi,
> 
> There a good point that @sftim makes in : https://kubernetes.slack.com/archives/CGW9JHPJN/p1559665692002800
> 
> Hope this helps!!!!!
> 
> Rajie

Hi Rajie,

Thanks a lot! Good point there. I replied on Slack and will work to add this info to the Kubernetes components section.

**Comment by shavidissa** [comment]
> @shavidissa This looks good!!!!
> my two cents: For https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#what-kubernetes-is-not
> 
> We could emphasis about what Kubernetes supports rather than what it does not.......
> Becasue, though we mention that Kubernetes does not.........we are also mentioning about the alternatives or the options that Kubernetes provides .....I would do thick on points that Kubernetes supports.
> I am not sure if we could change the title to " How Kubernetes supports common features in PaaS offerings" or something similar!!!!!

Good suggestion, @Rajakavitha1! Let me rename that heading .. that makes a lot of sense.

**Comment by shavidissa** [comment]
Hi @steveperry-53  and @juandiegopalomino,

Addressed your reviews and updated the content. Will you be able to check and give an lgtm if it is good to go, please?

**Comment by shavidissa** [comment]
Hi @steveperry-53,
If this looks ok, will you be able to _lgtm_ and _approve_, please? :)

**Pull request merged**